src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -83,7 +83,7 @@ public OpenApiDocument(OpenApiDocument document)
             Workspace = document?.Workspace != null ? new(document?.Workspace) : null;
             Info = document?.Info != null ? new(document?.Info) : null;
             Servers = document?.Servers != null ? new List<OpenApiServer>(document.Servers) : null;
-            Paths = document?.Paths != null ? new(document?.Paths) : null;
+            Paths = document?.Paths;
```

Comment:
I'm not sure why we're changing what we originally had. This will copy the reference which could have negative side effect (i.e. changing the dictionary content in the copy will impact the original).
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.-ctor?view=net-6.0#system-collections-generic-dictionary-2-ctor(system-collections-generic-idictionary((-0-1)))

```suggestion
            Paths = document?.Paths != null ? new(document.Paths) : null;
```

Reply:
We don't assign that parameter value to anything in the `OpenApiPaths` ctor: https://github.com/microsoft/OpenAPI.NET/blob/0b710d553eb5fe4bfceb0b1452923c09c6b2d098/src/Microsoft.OpenApi/Models/OpenApiPaths.cs#L19
The new document will always have its `Paths` empty

Reply:
ah I see, and that masks the copy constructor from dictionary. We should probably call the base constructor.
`public OpenApiPaths(OpenApiPaths paths): base(paths) {}`

And in `OpenApiExtensibleDictionary` have a copy constructor that calls the base, but also assigns the extensions.